packages/ui/src/composables/useTouch.ts

```diff
@@ -0,0 +1,88 @@
+import { ref, onBeforeUnmount, type Ref } from 'vue'
+
+import { useEvent, useWindow } from './'
+
+type UseTouchCallback = (e: TouchEvent) => void
+type UseTouchCallbacks = { short: UseTouchCallback, long: UseTouchCallback, double?: UseTouchCallback }
+
+/**
+ * Provides touch events handling.
+ *
+ * TODO: make a part of useMouseEvent (implement it first) hook.
+ *
+ * @param target - element that will be listened for touch start event.
+ * @param callbacks - callbacks for short, long and double touch events.
+ * @param ignoreDoubleClicks - if true, double clicks will be ignored.
+ * @param timings - timings for long and double touch events.
+ *
+ * @example
+ * ```ts
+ *  useTouch(target, {
+ *    short: () => {},
+ *    long: () => {},
+ *    double: () => {},
+ *    ignoreDoubleClicks: true,
+ *  })
+ *  ```
+ */
+export const useTouch = (
+  target: Ref<HTMLElement>,
+  callbacks: UseTouchCallbacks,
+  ignoreDoubleClicks = false,
+  timings = { long: 500, double: 250 },
+) => {
+  const window = useWindow()
+
+  if (!window.value || !('ontouchstart' in window.value)) { return }
+
+  const previousTouchTime = ref(0)
```

Comment:
There is nothing depends on this reactive variable, so it can be a let.
```suggestion
  let previousTouchTime = 0
```